clar2wasm/src/standard/standard.wat

```diff
@@ -741,6 +741,112 @@
+          (local $shift_lo i64)
+          (local $overflow i64)
+          (local $by i64)
+
+          (call $mod-uint (local.get $b_lo) (local.get $b_hi) (i64.const 128) (i64.const 0))
+          drop
+          (local.set $by)
+
+          ;; case when not shifting at all
+          (if (i64.eq (local.get $by)
+                      (i64.const 0))
+              (return (local.get $a_lo)
+                      (local.get $a_hi)))
+
+          ;; case when completely discarding high bits
+          (if (i64.ge_u (local.get $by) (i64.const 64))
+              (return
+                (i64.shr_u (local.get $a_hi)
+                           (i64.sub (local.get $by)
+                                    (i64.const 64)))
+                                (i64.const 0)))
+
+          ;; all other cases
+          (local.set $overflow (i64.shl (local.get $a_hi)
+                                        (i64.sub (i64.const 64)
+                                                 (local.get $by))))
+          ;; do the shifting, or-ing the overflow from hi into lo
+          (local.set $shift_hi (i64.shr_u (local.get $a_hi) (local.get $by)))
+          (local.set $shift_lo (i64.or (i64.shr_u (local.get $a_lo) (local.get $by))
+                                       (local.get $overflow)))
+          (return (local.get $shift_lo) (local.get $shift_hi)))
+
+    (func $bit-shift-right-int (type 1) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)
+          (local $shift_hi i64)
+          (local $shift_lo i64)
+          (local $overflow i64)
+          (local $by i64)
+
+          (call $mod-uint (local.get $b_lo) (local.get $b_hi) (i64.const 128) (i64.const 0))
+          drop
+          (local.set $by)
+
+          ;; case when not shifting at all
+          (if (i64.eq (local.get $by)
+                      (i64.const 0))
+              (return (local.get $a_lo)
+                      (local.get $a_hi)))
+
+          ;; case when completely discarding high bits
+          (if (i64.ge_u (local.get $by) (i64.const 64))
+              (return
+                (i64.shr_s (local.get $a_hi)
+                           (i64.sub (local.get $by)
+                                    (i64.const 64)))
+                ;; keep the sign by arithmetically shifting out everything
+                (i64.shr_s (local.get $a_hi) (i64.const 63))))
+
+          ;; all other cases
+          (local.set $overflow (i64.shl (local.get $a_hi)
+                                        (i64.sub (i64.const 64)
+                                                 (local.get $by))))
+          ;; do the shifting, or-ing the overflow from hi into lo
+          (local.set $shift_hi (i64.shr_s (local.get $a_hi) (local.get $by)))
+          (local.set $shift_lo (i64.or (i64.shr_u (local.get $a_lo) (local.get $by))
+                                       (local.get $overflow)))
+          (return (local.get $shift_lo) (local.get $shift_hi)))
```

Comment:
Any possibility for you to adapt this function so that it looks more like $bit_shift_left?

Reply:
done